src/memory/memory_chunk.cpp

```diff
@@ -71,7 +71,7 @@ void initializeUmpireDeviceAllocator(std::size_t initial_bytes) {
     auto device_allocator = umpire::ResourceManager::getInstance().getAllocator("DEVICE");
     auto pooled_device_allocator =
         umpire::ResourceManager::getInstance().makeAllocator<umpire::strategy::QuickPool>(
-            "DEVICE_pool", device_allocator, initial_bytes);
+            "DEVICE_pool", device_allocator, initial_bytes, std::size_t(1) << 30, std::size_t(1) << 21);
```

Comment:
No magic numbers. Currently this means:
- When growing the pool after the initial allocation, always grow by at least 1GiB.
- Alignment is 2MiB (the large page size). Using the regular page size (64KiB) for alignment improves performance, but doesn't completely remove the strange slowdowns.